### ISSUE TITLE
Automatic saving&loading of object properties.

### ISSUE THREAD

**Issue by Tachiorz** [aura-project/melia#53]
TLDR:
- save and load various object properties uniformly;
- automatically sync object properties to client.

Various objects in TOS have single parent with global unique ID and support for special properties. 
Those properties have id's and names and used to synchronize server side state of objects to client (I guess automatically). Examples in game protocol:
https://github.com/aura-project/melia/blob/bd83f147e4f50908dbdc306afdeb99eb90153854/src/LoginServer/Network/Helpers/AccountHelper.cs#L22
https://github.com/aura-project/melia/blob/9bdf2d8e579e0b9867f34358665c060f60824bb7/src/ChannelServer/Network/ChannelPacketSender.cs#L425
https://github.com/aura-project/melia/blob/9bdf2d8e579e0b9867f34358665c060f60824bb7/src/ChannelServer/Network/ChannelPacketSender.cs#L746

I suggest to do the same for melia.
Mark class properties with special attribute and they'll automatically sync to client on next `ZC_OBJECT_PROPERTY` (or any other packet that sends them). 
example:

```
/// <summary>
/// Health points.
/// </summary>
[PropertyAttribute(ObjectProperty.PC.HP)]
public int Hp { get; set; }

/// <summary>
/// Maximum health points.
/// </summary>
[PropertyAttribute(ObjectProperty.PC.MHP)]
public int MaxHp { get; set; }
```

Also we'll be able to save and load them uniformly using JSON (or protobuf?) This will save us the need to change DB structure often and writing tons of boilerplate code.

Proof of concept code https://github.com/Tachiorz/melia/tree/properties_delegates_test It looks nasty weird naming and scary code, so I'm going to refactor it.

**Comment by exectails** [comment]
I'd wish we could solve the caching some other way, but I can't think of anything. Other than that I really like this idea. Btw, you don't have to include the `Attribute` suffix when you use the class as an attribute, the compiler will figure that out itself.

```csharp
[Property(ObjectProperty.PC.MHP)]
public int MaxHp { get; set; }
```

**Comment by exectails** [comment]
I'm closing this, because Tachi isn't around anymore, and we have a different property system in place now, which isn't directly compatible with the original idea.
